src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -167,6 +167,29 @@ public SnapshotService(
     this.snapshotBuilderSettingsDao = snapshotBuilderSettingsDao;
   }
 
+  public String generateUpdatedSnapshotNameIfByRequest(SnapshotRequestModel model) {
+    SnapshotRequestContentsModel contentsModel = model.getContents().get(0);
+    if (contentsModel.getMode() == SnapshotRequestContentsModel.ModeEnum.BYREQUESTID) {
+      SnapshotAccessRequestResponse snapshotAccessRequestResponse =
+          snapshotRequestDao.getById(contentsModel.getRequestIdSpec().getSnapshotRequestId());
+      String dashesAndSpacesRegex = "[- ]+";
+      String nonAlphaNumericRegex = "\\W";
+
+      return org.springframework.util.StringUtils.trimTrailingCharacter(
```

Comment:
It would be nice to avoid using the spring StringUtils here since that isn't an API used elsewhere in TDR. E.g., for removing start and end text, apache `StringUtils.strip(x, y)`.

Also, this code mixes together the code that sanitizes the name in the request with the code that builds the new name. I think it would be better to first sanitize the user entered name, and then build the new name. Something like

```
  var requestName = accessRequest.getSnapshotName();
  var cleanedName = requestName.replaceAll(...);
  var generatedName = "%s_%s".formatted(cleanedName, accessRequest.getId());
```

Reply:
We also need to swap the `-` in the id for `_`, so that also needs to be sanitized.

Reply:
Given those cases, maybe a more specific approach would be clearer here
```
  var requestName = accessRequest.getSnapshotName();
  var cleanedName = requestName != null ? requestName.replaceAll(...) + "_": "";
  var generatedName = cleanedName + accessRequest.getId().toString().replace('-', '_');
```

Reply:
Okay, I will clean it up and separate it all out.